classes/plugins/PubObjectsExportPlugin.inc.php

```diff
@@ -0,0 +1,523 @@
+		if (!file_exists($exportPath)) {
+			$fileManager = new FileManager();
+			$fileManager->mkdir($exportPath);
+		}
+		if (!is_writable($exportPath)) {
+			$errors = array(
+				array('plugins.importexport.common.export.error.outputFileNotWritable', $exportPath)
+			);
+			return $errors;
+		}
+		return realpath($exportPath) . '/';
+	}
+
+	/**
+	 * Return the whole export file name.
+	 * @param $objectsFileNamePart string Part different for each object type.
+	 * @param $context Context
+	 * @return string
+	 */
+	function getExportFileName($objectsFileNamePart, $context) {
+		return $this->getExportPath() . date('Ymd-His') .'-' . $objectsFileNamePart .'-' . $context->getId() . '.xml';
+	}
+
+	/**
+	 * Remove the given temporary file.
+	 * @param $tempfile string
+	 */
+	function cleanTmpfile($tempfile) {
+		if (file_exists($tempfile)) {
+			unlink($tempfile);
+		}
+	}
+
+	/**
+	 * Mark selected submissions or issues as registered.
+	 * @param $context Context
+	 * @param $objects array Array of published articles, issues or galleys
+	 */
+	function markRegistered($context, $objects) {
+		foreach ($objects as $object) {
+			$object->setData($this->getDepositStatusSettingName(), EXPORT_STATUS_MARKEDREGISTERED);
+			$this->updateObject($object);
+		}
+	}
+
+	/**
+	 * Update the given object.
+	 * @param $object Issue|PublishedArticle|ArticleGAlley
+	 */
+	function updateObject($object) {
+		// Register a hook for the required additional
+		// object fields. We do this on a temporary
+		// basis as the hook adds a performance overhead
+		// and the field will "stealthily" survive even
+		// when the DAO does not know about it.
+		$dao = $object->getDAO();
+		$this->registerDaoHook(get_class($dao));
+		$dao->updateObject($object);
+	}
+
+	/**
+	 * Register the hook that adds an
+	 * additional field name to objects.
+	 * @param $daoName string
+	 */
+	function registerDaoHook($daoName) {
+		HookRegistry::register(strtolower_codesafe($daoName) . '::getAdditionalFieldNames', array(&$this, 'getAdditionalFieldNames'));
+	}
+
+	/**
+	 * Hook callback that returns the setting's name prefixed with
+	 * the plug-in's id to avoid name collisions.
+	 * @see DAO::getAdditionalFieldNames()
+	 * @param $hookName string
+	 * @param $args array
+	 */
+	function getAdditionalFieldNames($hookName, $args) {
+		assert(count($args) == 2);
+		$additionalFields =& $args[1];
+		assert(is_array($additionalFields));
+		$additionalFields[] = $this->getDepositStatusSettingName();
+	}
+
+	/**
+	 * Check whether we are in test mode.
+	 * @param $context Context
+	 * @return boolean
+	 */
+	function isTestMode($context) {
+		return ($this->getSetting($context->getId(), 'testMode') == 1);
+	}
+
+	/**
+	 * Get deposit status setting name.
+	 * @return string
+	 */
+	function getDepositStatusSettingName() {
+		return $this->getPluginSettingsPrefix().'::status';
+	}
+
+
+
+	/**
+	 * @copydoc PKPImportExportPlugin::usage
+	 */
+	function usage($scriptName) {
```

Comment:
May I leave them for now, because the child classes don't have them implemented for now?